__tests__/index.js

```diff
@@ -186,3 +186,63 @@ test('check if hierarchyParser is ephemeral', t => {
   t.deepEqual(hierarchyOne, expected)
   t.deepEqual(hierarchyTwo, expected)
 })
+
+test('check if hierarchyParser does not mutate the input data', (t) => {
+  const f = Object.freeze.bind(null)
```

Comment:
why not `freeze` instead of `f` ?

Reply:
Because my intention is to create an alias to write code here in a more versatile way